### PR TITLE
An option to retry a failed service; test on [dynamicxml]

### PR DESCRIPTION
**Pull request by platan**
This PR adds an option to retry all failed service tests. It uses IcedFrisby's [retry()](https://github.com/IcedFrisby/IcedFrisby/blob/master/API.md#retrycount-backoff) function. 
Usage:
```bash
RETRY_COUNT=3 RETRY_BACKOFF=0 npm run test:services -- --only=dynamicxml
```
Retried tests are marked with `R` in logs. An example:
```
    [live] XML from url | invalid url
RR      ✓ 
	[ GET .json?url=https://github.com/badges/shields/raw/master/notafile.xml&query=//version ] (413ms)
```
Default values:
- count: 0
- backoff: 0

I've tested it manually by setting different values of a timeout for specific service tests. 
I would like to use the functionality from this PR for https://github.com/badges/shields/issues/2688.

### CODE REVIEW

File: core/service-test-runner/service-tester.js [hunk 1]
```diff
@@ -116,7 +116,7 @@ class ServiceTester {
    * @param {string} attrs.baseUrl base URL for test server
    * @param {boolean} attrs.skipIntercepted skip tests which intercept requests
```
Comment (calebcartwright) [inline core/service-test-runner/service-tester.js]: I don't recall offhand how to best handle a nested object param, but we should probably update the jsdoc here to account for the new retry param

File: core/service-test-runner/cli.js [hunk 2]
```diff
@@ -15,6 +15,11 @@
 // Run tests on a given instance:
 //   SKIP_INTERCEPTED=TRUE TESTED_SERVER_URL=https://test.shields.io npm run test:services --
 //
+// Run tests with given number of retries and backoff:
+//   RETRY_COUNT=3 RETRY_BACKOFF=100 npm run test:services --
```
Comment (PyvesB) [inline core/service-test-runner/cli.js]: Shall we indicate that it's in milliseconds here?

File: core/service-test-runner/cli.js [hunk 3]
```diff
@@ -59,6 +64,9 @@ const Runner = require('./runner')
 
 require('../unhandled-rejection.spec')
 
+const retry = {}
+retry.count = process.env.RETRY_COUNT || 0
```
Comment (calebcartwright) [inline core/service-test-runner/cli.js]: Will these need to be explicitly cast to numbers/integers? I'm not familiar with IcedFrisby's retry, but I've run into issues with env vars being stringified and then libs that are expecting numbers bark at those values

**Comment by paulmelnikow** [comment]
This is awesome! Thanks so much for taking this on, @platan!

Review seems well under control though I do have one question. Will we set these env vars for the service tests in CI and daily tests? Should we turn on the retries on by default so they run in dev too?

File: core/service-test-runner/service-tester.js (hunk 1 above)
Comment (platan) [reply core/service-test-runner/service-tester.js]: Thanks! https://jsdoc.app/tags-param.html describes this case:

> If a parameter is expected to have a specific property, you can document that property by providing an additional @param tag. For example, if an employee parameter is expected to have name and department properties, you can document it as follows:
> Documenting a parameter's properties
> 
> /**
>  * Assign the project to an employee.
>  * @param {Object} employee - The employee who is responsible for the project.
>  * @param {string} employee.name - The name of the employee.
>  * @param {string} employee.department - The employee's department.
>  */
> Project.prototype.assign = function(employee) {
>     // ...
> };

I expected ESLint+JSDoc plugin to catch it. But JSDoc plugin doesn't support a destructuring parameter: https://github.com/gajus/eslint-plugin-jsdoc/issues/11

File: core/service-test-runner/cli.js (hunk 2 above)
Comment (platan) [reply core/service-test-runner/cli.js]: Thanks. Done.

File: core/service-test-runner/cli.js (hunk 3 above)
Comment (platan) [reply core/service-test-runner/cli.js]: [Documentation](https://github.com/IcedFrisby/IcedFrisby/blob/master/API.md#retrycount-backoff) says that count and backoff should be numbers, but `retry` works with strings. Nevertheless, I converted it to integers.

**Comment by platan** [comment]
@paulmelnikow I wanted to start with retries in daily tests. But we can add them to service tests in CI as well. 
But I think we shouldn't turn it on in dev. If you get a timeout (it's the only error after which retry can help) you should probably increase the default timeout and rerun the test manually.

**Comment by paulmelnikow** [comment]
That sounds good to me!

**Pull request merged**